### DISCUSSION TITLE
Do something when node size smaller than threshold

### DISCUSSION THREAD

**Discussion by rookie-orange** [vasturiano/react-force-graph#549] [Q&A]
I want to do something when the node size is smaller than a certain threshold. What do I do?

**Comment by vasturiano** [comment]
@rookie-orange thanks for reaching out.

Could you elaborate a bit in your question of what you're trying to achieve? So that we can provide a more useful answer.

**Reply by rookie-orange** [reply]
Thank you very much!
In my code, I generated a label for each node to show their name.  
```typescript
.nodeThreeObject((node) => {
  const nodeEl = document.createElement('div');
  nodeEl.textContent = node.id as string;
  nodeEl.className = 'text-xs p-1 rounded-md bg-background/20';
  return new CSS2DObject(nodeEl);
})
```
However, this can cause performance problems when the number of nodes is large, and I want only the nodes that look large enough to display the label.
I look forward to and appreciate your reply.

**Reply by vasturiano** [reply]
@rookie-orange for the nodes you want to fall back to default representation you can just return `null` in the `nodeThreeObject` method. Like:
```js
.nodeThreeObject(node => {
  if (isNodeTooSmall(node)) { // implementation of isNodeTooSmall is up to you
    return null;
  }
  // rest of code
})
```
